### PR TITLE
Convert testOptimizationProblem to use googletest

### PR DESCRIPTION
**Pull request by sammy-tri**

### CODE REVIEW

File: drake/solvers/test/testOptimizationProblem.cpp [hunk 1]
```diff
@@ -1,5 +1,8 @@
 
 #include <typeinfo>
+
+#include <gtest/gtest.h>
```
Comment (david-german-tri) [inline drake/solvers/test/testOptimizationProblem.cpp]: does cpplint accept this?  I think you need quotes.

File: drake/solvers/test/testOptimizationProblem.cpp [hunk 2]
```diff
@@ -252,28 +257,10 @@ void multiLCP() {
...
-    // and proceed; if not, reraise it to fail.
-    if (std::string(e.what()) != "SNOPT unavailable") {
-      throw;
-    }
-  }
-  simpleLCPConstraintEval();
-  simpleLCP();
-  multiLCP();
-  return 0;
 }
```
Comment (david-german-tri) [inline drake/solvers/test/testOptimizationProblem.cpp]: // namespace

(please run cpplint!)

Comment (sammy-tri) [inline drake/solvers/test/testOptimizationProblem.cpp]: drake-distro/drake% cpplint solvers/test/testOptimizationProblem.cpp
Done processing solvers/test/testOptimizationProblem.cpp
Total errors found: 0

I'm not clear on what you expected cpplint to output?  Is there a cpplint configuration I'm missing here?

Still need to look into the automated test issues.

Comment (david-german-tri) [inline drake/solvers/test/testOptimizationProblem.cpp]: No, my fault -  I thought this brace closed an anonymous namespace, but actually it closes main().

**Comment by sammy-tri** [comment]
I made a fairly significant error here in not thinking about optional SNOPT support.  I have a fix for this in my nlopt branch, but that's not ready to merge.

For now, I'm withdrawing this PR in favor of getting #1953 in.  I'll revisit the issues here another time.

**Pull request closed without merging**